src/operation/run_cursor_command.rs

```diff
@@ -96,24 +99,31 @@ impl<'conn> Operation for RunCursorCommand<'conn> {
         response: RawCommandResponse,
         description: &StreamDescription,
     ) -> Result<Self::O> {
-        let doc = Operation::handle_response(&self.run_command, response, description)?;
-        let cursor_info = bson::from_document(doc)?;
-        let batch_size = match &self.options {
-            Some(options) => options.batch_size.clone(),
-            None => None,
-        };
-        let max_time = match &self.options {
-            Some(options) => options.max_time.clone(),
-            None => None,
-        };
+        let cursor_response: CursorBody = response.body()?;
+
         let comment = match &self.options {
             Some(options) => options.comment.clone(),
             None => None,
         };
+
+        let mut max_time = Some(Duration::new(0, 0));
+
+        if let Some(ref options) = self.options {
+            match options.cursor_type {
+                Some(CursorType::Tailable) | Some(CursorType::TailableAwait) => {
+                    match &self.options {
+                        Some(options) => options.max_time,
+                        None => None,
+                    };
+                }
+                _ => max_time = None,
+            };
+        }
```

Comment:
I think (based on [this test](https://github.com/mongodb/mongo-rust-driver/blob/76cf34921fc23facce470e4aef253bb5d88ae003/src/test/spec/json/run-command/unified/runCursorCommand.json#L575-L576) that I suggested unskipping below) we actually don't need to worry about the user's cursor type here. A cursor type is not configured in that call to `run_cursor_command` and the `max_time` field is still expected to be attached to the calls to getMore.